### PR TITLE
Current user based index data

### LINKED ISSUE #204: Fix user-skills and user-categories index actions to return authenticated user data
Right now the `index` actions in both `user-skills` and `user-categories` return everyone's data, instead of the authenticated user's when authenticated.

In Rails we did the following:

```
    if id_params.present?
      user_skills = UserSkill.includes(:skill, :user).where(id: id_params)
    elsif current_user.present?
      user_skills = UserSkill.includes(:skill, :user).where(user_id: current_user.id)
    else
      user_skills = []
    end
```

We need to fix this the `filter` or find some other approach to correcting the data returned from this endpoint.

### PR DESCRIPTION
**Pull request by npendery**
Closes #204 

On top of @JoshSmith's #207

### CODE REVIEW

File: web/controllers/user_skill_controller.ex [hunk 1]
```diff
@@ -10,6 +10,17 @@ defmodule CodeCorps.UserSkillController do
...
+      case conn.assigns[:current_user] do
+        nil ->
+          params
+        %CodeCorps.User{id: user_id} ->
+          Map.put(params, :current_user_id, user_id)
+        _ ->
+          params
+      end
+
+    Kernel.inspect(params) |> IO.puts
```
Comment (joshsmith) [inline web/controllers/user_skill_controller.ex]: Is this supposed to be here?

File: web/plugs/current_user.ex [hunk 2]
```diff
@@ -1,20 +1,22 @@
 defmodule CodeCorps.Plug.CurrentUser do
+  @analytics Application.get_env(:code_corps, :analytics)
```
Comment (JoshSmith) [inline web/plugs/current_user.ex]: How did analytics code make it in here?

**Review by joshsmith — CHANGES_REQUESTED** [review]
I think doing this on top of #207 wasn't a good idea, because you've had to undo changes in there. I would have just pulled in the auth-type changes I was making there, or do that as a separate PR and I could've rebased from there.

I would consider doing that now since I'm not sure what effect this will have on that existing PR and the commit history.

**Comment by begedin** [comment]
@npendery 

Sorry about this, but it seems we'll have to close the PR without merging. We wanted to mirror the rails api here, but once we took another good look, we concluded it doesn't make sense to do it that way.

Here's what rails does:

``` Ruby
if id_params.present?
  user_skills = UserSkill.includes(:skill, :user).where(id: id_params)
elsif current_user.present?
  user_skills = UserSkill.includes(:skill, :user).where(user_id: current_user.id)
else
  user_skills = []
end
```

To put it in words, if the client provides a list of ids it wants, they get that back. If they don't, then if they are authenticated, they get all of their own records back.

The reality is, they shouldn't need to explicitly fetch their own records. They should get a list of ids with their user payload, and then simply fetch those records using these ids.

If there anything, we might eventually need a `user_filter`instead of a `current_user_filter`, meaning the client would provide a `user_id` to filter by. For now, though, we can close this and see if there's anything that needs fixing client-side.

**Pull request closed without merging**